frontend/src/pages/modelRegistry/screens/RegisteredModels/useRegisteredModelContext.ts

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import useModelArtifactsByVersionId from '~/concepts/modelRegistry/apiHooks/useModelArtifactsByVersionId';
+import useRegisteredModelById from '~/concepts/modelRegistry/apiHooks/useRegisteredModelById';
+import { ModelVersion } from '~/concepts/modelRegistry/types';
+
+export type RegisteredModelContext = {
+  modelName: string;
+  modelFormat?: string;
+  modelArtifactUri?: string;
+  modelArtifactStorageKey?: string;
+};
+
+const useRegisteredModelContext = (modelVersion: ModelVersion): RegisteredModelContext => {
+  const [registeredModel] = useRegisteredModelById(modelVersion.registeredModelId);
+  const [modelArtifactList] = useModelArtifactsByVersionId(modelVersion.id);
```

Comment:
Hmm, I am not sure, the only side-effect here is if there is an error, the modal will not pre-fill those values, but that doesn't block users from deploying the model, they just need to select by themselves. I am not sure the best way to handle the loading and error state here, WDYT?

Reply:
I suppose the user can't get to this deployment action unless they've already gone through the Model/Version details pages which will be making these same API requests, so it's not critical to catch errors here - but technically if the user e.g. loses their internet connection and then clicks Deploy, we should ideally show an error instead of failing silently.

Can we add two more properties to `RegisteredModelDeployInfo`:
```ts
loaded: boolean;
loadError?: Error;
```
and where we call `useRegisteredModelById` and `useModelArtifactsByVersionId`, can we pull out the `loaded` and `loadError` values from those returned arrays, and then when we return `RegisteredModelDeployInfo` we can aggregate them like:
```ts
loaded: registeredModelLoaded && modelArtifactsLoaded,
loadError: registeredModelLoadError || modelArtifactsLoadError,
```
and then in the `DeployRegisteredModelModal`, in the body of the modal can we render an Alert if there is a `loadError` in that object or a Spinner if it's `!loaded`?